src/main/java/org/commonjava/util/sidecar/services/ArchiveRetrieveService.java

```diff
@@ -0,0 +1,279 @@
+                                                         {
+                                                             logger.info( "NoHttpResponse start to retry times:"
+                                                                                          + executionCount );
+                                                             return true;
+                                                         }
+                                                         if ( exception instanceof SocketTimeoutException )
+                                                         {
+                                                             logger.info( "SocketTimeout start to retry times:"
+                                                                                          + executionCount );
+                                                             return true;
+                                                         }
+                                                         if ( exception instanceof ConnectionPoolTimeoutException )
+                                                         {
+                                                             logger.info( "ConnectionPoolTimeout start to retry times:"
+                                                                                          + executionCount );
+                                                             return true;
+                                                         }
+                                                         return false;
+                                                     } );
+        client = builder.build();
+    }
+
+    @PreDestroy
+    public void destroy()
+    {
+        try
+        {
+            File downloadDir = new File( localRepository.get() );
+            List<File> downloads = Files.walk( downloadDir.toPath() )
+                                        .filter( Files::isRegularFile )
+                                        .map( Path::toFile )
+                                        .collect( Collectors.toList() );
+            for ( File content : downloads )
+            {
+                content.delete();
+            }
+            downloadDir.delete();
+        }
+        catch ( IOException e )
+        {
+            logger.error( "Failed to clean up downloads", e );
+        }
+        finally
+        {
+            IOUtils.closeQuietly( client, null );
+        }
+    }
+
+    public boolean decompressArchive()
+    {
+        if ( archiveApi.isEmpty() )
+        {
+            return false;
+        }
+        String buildConfigId = System.getProperty( BUILD_CONFIG_ID );
+        final File target = new File( localRepository.orElse( DEFAULT_REPO_PATH ), buildConfigId + ARCHIVE_SUFFIX );
+
+        if ( !retrieveArchive( target ) )
+        {
+            return false;
+        }
+        if ( target.exists() && target.length() <= 0 )
```

Comment:
So what will happen if !target.exists() ?

Reply:
@ligangty target is new in #L162, it will have no existed case but after retrieving the archive and stream writing, it will have the case of length=0, so hold here.